### PR TITLE
Add sp_between_semi_for_empty option

### LINKED ISSUE #3280: Need an option to add spaces around "forever" loops but not between the semicolons
Config file:

```
sp_before_semi_for_empty = add
sp_after_semi_for_empty = add
```

Example code:

```
int main(void)
{
	for (; ;)
		continue;
}
```

Output:

```
int main(void)
{
	for ( ; ; )
		continue;
}
```

What I'm looking for:

```
int main(void)
{
	for ( ;; )
		continue;
}
```

So, Uncrustify has the options to add a space before the first semicolon and after the second semicolon, but it has no option to remove the space in the middle.

### PR DESCRIPTION
**Pull request by alexhenrie**
Fixes #3280

### CODE REVIEW

**Review by gmaurel — COMMENTED** [review]
have a look to
https://github.com/uncrustify/uncrustify/issues/3280#issuecomment-900392451

**Review by gmaurel — CHANGES_REQUESTED** [review]
Fine, but we need to test the case if 
 ` options::sp_between_semi_for_empty() == IARF_IGNORE`

**Comment by alexhenrie** [comment]
Done. Please let me know if there's anything else I should change.

File: src/space.cpp [hunk 1]
```diff
@@ -412,6 +412,14 @@ static iarf_e do_space(chunk_t *first, chunk_t *second, int &min_sp)
 
       if (get_chunk_parent_type(second) == CT_FOR)
       {
+         if (  chunk_is_token(first, CT_SEMICOLON)
+            && options::sp_between_semi_for_empty() != IARF_IGNORE)
+         {
+            // Add or remove space between semicolons of an empty middle part of a for statement.
+            log_rule("sp_between_semi_for_empty");
+            return(options::sp_between_semi_for_empty());
+         }
```
Comment (gmaurel) [inline src/space.cpp:421]: we need here some code to consider
`options::sp_between_semi_for_empty() == IARF_IGNORE`

Comment (alexhenrie) [reply src/space.cpp:423]: OK, I have modified the patch so that sp_before_semi_for_empty and sp_between_semi_for_empty function independently.

**Review by gmaurel — APPROVED** [review]

**Comment by gmaurel** [comment]
Fine, thanks

**Pull request merged**